resharper/resharper-unity/src/CSharp/Daemon/Stages/Analysis/PropertyDrawerOnGUIBaseProblemAnalyzer.cs

```diff
@@ -0,0 +1,85 @@
+using JetBrains.ReSharper.Feature.Services.Daemon;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Errors;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Stages.Dispatcher;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.Impl.Reflection2;
+using JetBrains.ReSharper.Psi.Resolve;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Stages.Analysis
+{
+    [ElementProblemAnalyzer(typeof(IInvocationExpression), HighlightingTypes =
+        new[] {typeof(PropertyDrawerOnGUIBaseWarning)})]
+    public class PropertyDrawerOnGUIBaseProblemAnalyzer : UnityElementProblemAnalyzer<IInvocationExpression>
+    {
+        public PropertyDrawerOnGUIBaseProblemAnalyzer(UnityApi unityApi)
+            : base(unityApi)
+        {
+        }
+
+        protected override void Analyze(IInvocationExpression expression, ElementProblemAnalyzerData data,
+            IHighlightingConsumer consumer)
+        {
+            if (IsOnGUIBaseCall(expression) &&
+                IsInsidePropertyDrawer(expression) &&
+                IsInsideOnGUI(expression))
+            {
+                var invocation = (expression.InvokedExpression as IReferenceExpression)?.Reference;
+                consumer.AddHighlighting(new PropertyDrawerOnGUIBaseWarning(invocation));
+            }
+        }
+
+        private static bool IsOnGUIBaseCall(IInvocationExpression expression)
+        {
+            var reference = expression.Reference;
+            if (reference == null)
+                return false;
+
+            var info = reference.Resolve();
+
+            if (info.ResolveErrorType != ResolveErrorType.OK)
+                return false;
+
+            var method = (info.DeclaredElement as IMethod).NotNull("info.DeclaredElement as IMethod != null");
```

Comment:
You can't use `NotNull` here. This is an assert and will throw if the expression is invoking something else (maybe a property or indexer, etc).